File: Algorithms/Warmup/Time Conversion/TimeConversion.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    static String timeConversion(String s) {
        String[] timeParts = s.split(":");
        int hours = Integer.parseInt(timeParts[0]);
        int minutes = Integer.parseInt(timeParts[1]);
        int seconds = Integer.parseInt(timeParts[2].substring(0, 2));
        boolean pm = timeParts[2].substring(2).equals("PM");
        if (hours == 12) {
            hours = pm ? 12 : 0;
        } else if (pm) {
            hours += 12;
            if (hours > 23) {
                hours -= 24;
            }
        }
        return (hours < 10 ? "0" + hours : hours) + ":" + (minutes < 10 ? "0" + minutes : minutes) + ":" + (seconds < 10 ? "0" + seconds : seconds);
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String s = in.next();
        String result = timeConversion(s);
        System.out.println(result);
    }
}

File: Algorithms/Warmup/Time Conversion/TimeConversion.py
#!/bin/python3

import sys

def timeConversion(s):
    timeParts = s.split(":")
    hours = int(timeParts[0])
    minutes = int(timeParts[1])
    seconds = int(timeParts[2][:2]);
    pm = timeParts[2][2:] == "PM"
    if hours == 12:
        hours = 12 if pm else 0
    elif pm:
        hours += 12
        if hours > 23:
            hours -= 24
    return ("0" + str(hours) if hours < 10 else str(hours)) + ":" + ("0" + str(minutes) if minutes < 10 else str(minutes)) + ":" + ("0" + str(seconds) if seconds < 10 else str(seconds))

s = input().strip()
result = timeConversion(s)
print(result)
